tests/test_grid.py

```diff
@@ -52,12 +52,20 @@ def test_make_square_grid():
 
     test = CoreGrid._make_square_grid(100, 10, 10)
 
-    # TODO check if correct number of points were created
-
-    # TODO check that distance between neighboring points is euqal for all integers
-    distance = euclidean(test["0-0"]["poly"][0], test["0-0"]["poly"][1])
-    distance2 = euclidean(test["0-0"]["poly"][0], test["0-0"]["poly"][3])
-    assert distance == distance2
+    # check if correct number of points were created
+    assert len(test.keys()) == 100
+
+    # check that distance between neighboring points is euqal for all integers
+    # (this works but is a bit long)
+    for x in range(9):
```

Comment:
It is not necessary to test every possible combination to have some reassurance that this is correct. What I would do is to pick 3 random values for x and y within the [0, 9] range and check if this condition is fulfilled for them. That would be 10 times faster and I would say it is good enough.

Reply:
I implemented hypothesis, so no longer relevant. However, very useful to know that tests don't need to be so extensive :-)